MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+
+## Motivation
+
+The current bridge design poses numerous challenges, including inefficiency, cost, and user frustration:
+
+1. **Transaction Complexity**: Requires four interactions:
+   - Two by the user (initiation and finalization).
+   - One by the relayer to complete the transaction.
+   - One by a third party (tasked by the relayer) to finalize refunds in case of failure.
+   - If any of these fails, users and the protocol face losses.
+2. **High Cost**:
+   - The multi-transaction setup is expensive for both users and the protocol.
+   - Misestimations in fee calculations could result in significant losses.
+   - Example: At 16 gwei, bridging costs $20 for 10k transactions. If fees are miscalculated by 25%, this leads to a $25k loss. At 100 gwei, the loss escalates dramatically.
+3. **Unfriendly User Experience**:
+   - The current design requires users to have funds on the destination chain for finalization. This increases friction and discourages adoption.
+   - Sponsored transactions are essential to the current implementation but remain unimplemented, leaving users stranded without funds.
+4. **Security Risks**:
+   - Relayer keys are a critical point of failure. This assumption cannot be avoided.
+   - Refund logic introduces vulnerabilities where attackers could exploit the rate limit and relayer downtime.
+   - For instance, if the relayer fails to finalize on the initiator, a malicious actor could:
+     - Take over the refund keys.
+     - Exploit both directions of the bridge (e.g., ETH→MOVE and MOVE→ETH) repeatedly, draining funds.
+5. **Maintenance Burden**:
+   - The current audit is outdated and does not reflect the significant changes made since.
+   - The current design requires a complete UI/UX overhaul, adding complexity and delay.
+6. **Unnecessary Complexity**:
+   - HTLC-based bridges are largely abandoned in favor of simpler, more effective designs.
+   - Examples like the [Consensys HTLC](https://github.com/Consensys/htlc-bridge) bridge demonstrate the [pitfalls of such approaches](https://entethalliance.org/crosschain-bridges-overview-where-we-are-now/).
+7. **Infrastructure Simplification**:
+   - Infrastructure is still incomplete for the current HTLC bridge design and we are struggling to find sound solutions that balance UI/UX and security. For example, [this issue](https://github.com/movementlabsxyz/movement/issues/838) regarding UX vs number of confirmations to require on the Ethereum side, is still under debate.
```

Comment:
This issue does not relate to HTLC. This issue is not going to be resolved by a new architecture. In a cross chain "deal" (bridge, atomic swap etc) the two chains need to communicate their states to the other chain. The communicated states must be trusted by each party. 
In a trustless set up, the two chains may run a (light)-client of the other chain to verify the "validity" of a block/state (using Merkle proofs etc).
In a trusted set up, e.g., the relayer is trusted, the relayer is trusted to relay information faithfully (covering events to transactions). But we have to make sure that the information about the state is trustworthy i.e. that the states that are communicated (relayed) by the relayer are final or not going to be reverted (re-orgd) .

In the current state of Ethereum we know that:
1. re-org occurs approximately once every hour
2. no-reorg on finalised blocks.

In between (k-confirmations) we don't have data or evidence that it is safe to use the blocks/states. 
If we choose this route, we have to take into account possible re-orgs which arguably complicates the bridge design.